pkg/clusterapi/registryMirror_test.go

```diff
@@ -0,0 +1,125 @@
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
+    endpoint = ["https://1.2.3.4:443"]
+  [plugins."io.containerd.grpc.v1.cri".registry.configs."1.2.3.4:443".tls]
+    ca_file = "/etc/containerd/certs.d/1.2.3.4:443/ca.crt"`,
+			},
+			{
+				Path:    "/etc/containerd/certs.d/1.2.3.4:443/ca.crt",
+				Owner:   "root:root",
+				Content: "xyz",
+			},
+		},
+	},
+	{
+		name: "with insecure skip",
+		registryMirrorConfig: &v1alpha1.RegistryMirrorConfiguration{
+			Endpoint:           "1.2.3.4",
+			Port:               "443",
+			InsecureSkipVerify: true,
+		},
+		wantFiles: []bootstrapv1.File{
+			{
+				Path:  "/etc/containerd/config_append.toml",
+				Owner: "root:root",
+				Content: `[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
+    endpoint = ["https://1.2.3.4:443"]
+  [plugins."io.containerd.grpc.v1.cri".registry.configs."1.2.3.4:443".tls]
+    insecure_skip_verify = true`,
+			},
+		},
+	},
+	{
+		name: "with ca cert and insecure skip",
+		registryMirrorConfig: &v1alpha1.RegistryMirrorConfiguration{
+			Endpoint:           "1.2.3.4",
+			Port:               "443",
+			CACertContent:      "xyz",
+			InsecureSkipVerify: true,
+		},
+		wantFiles: []bootstrapv1.File{
+			{
+				Path:  "/etc/containerd/config_append.toml",
+				Owner: "root:root",
+				Content: `[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
+    endpoint = ["https://1.2.3.4:443"]
+  [plugins."io.containerd.grpc.v1.cri".registry.configs."1.2.3.4:443".tls]
+    ca_file = "/etc/containerd/certs.d/1.2.3.4:443/ca.crt"
+    insecure_skip_verify = true`,
+			},
+			{
+				Path:    "/etc/containerd/certs.d/1.2.3.4:443/ca.crt",
+				Owner:   "root:root",
+				Content: "xyz",
+			},
+		},
+	},
+}
+
+func wantRegistryMirrorCommands() []string {
+	return []string{
+		"cat /etc/containerd/config_append.toml >> /etc/containerd/config.toml",
+		"sudo systemctl daemon-reload",
+		"sudo systemctl restart containerd",
+	}
+}
+
+func TestKubeadmControlPlaneWithRegistryMirror(t *testing.T) {
+	for _, tt := range registryMirrorTests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := newApiBuilerTest(t)
+			got := wantKubeadmControlPlane()
+			err := clusterapi.KubeadmControlPlaneRegistryMirror(got, tt.registryMirrorConfig)
+			g.Expect(err).To(Succeed())
+			want := wantKubeadmControlPlane()
+			want.Spec.KubeadmConfigSpec.Files = tt.wantFiles
+			want.Spec.KubeadmConfigSpec.PreKubeadmCommands = wantRegistryMirrorCommands()
+			g.Expect(got).To(Equal(want))
+		})
+	}
+}
+
+func TestKubeadmConfigTemplateWithRegistryMirror(t *testing.T) {
+	for _, tt := range registryMirrorTests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := newApiBuilerTest(t)
+			got := wantKubeadmConfigTemplate()
+			err := clusterapi.KubeadmConfigTemplateRegistryMirror(got, tt.registryMirrorConfig)
+			g.Expect(err).To(Succeed())
```

Comment:
nit: it could be a bit more idiomatic like
```suggestion
			g.Expect(clusterapi.KubeadmConfigTemplateRegistryMirror(got, tt.registryMirrorConfig)).To(Succeed())
```